tests/processor/document/test_candidate_relation_adder.py

```diff
@@ -0,0 +1,347 @@
+import copy
+import dataclasses
+
+import pytest
+from pytorch_ie.annotations import BinaryRelation, LabeledSpan, Span
+from pytorch_ie.core import AnnotationList, annotation_field
+from pytorch_ie.documents import TextDocument
+
+from src.processor.document.candidate_relation_adder import CandidateRelationAdder
+
+TEXT_01 = "Jane lives in Berlin. this is no sentence about Karl\n"
+TEXT_02 = "Seattle is a rainy city. Jenny Durkan is the city's mayor.\n"
+TEXT_03 = "Karl enjoys sunny days in Berlin."
+
+ENTITY_TEXT_01_JANE = LabeledSpan(start=0, end=4, label="person")
+ENTITY_TEXT_01_BERLIN = LabeledSpan(start=14, end=20, label="city")
+ENTITY_TEXT_01_KARL = LabeledSpan(start=48, end=52, label="person")
+TEXT_01_SENTENCE1 = LabeledSpan(start=0, end=21, label="sentence")
+REL_JANE_LIVES_IN_BERLIN = BinaryRelation(
+    head=ENTITY_TEXT_01_JANE, tail=ENTITY_TEXT_01_BERLIN, label="lives_in"
+)
+
+ENTITY_TEXT_02_SEATTLE = LabeledSpan(start=0, end=7, label="city")
+ENTITY_TEXT_02_JENNY = LabeledSpan(start=25, end=37, label="person")
+TEXT_02_SENTENCE1 = LabeledSpan(start=0, end=24, label="sentence")
+TEXT_02_SENTENCE2 = LabeledSpan(start=25, end=58, label="sentence")
+REL_JENNY_MAYOR_OF_SEATTLE = BinaryRelation(
+    head=ENTITY_TEXT_02_JENNY, tail=ENTITY_TEXT_02_SEATTLE, label="mayor_of"
+)
+
+ENTITY_TEXT_03_KARL = LabeledSpan(start=0, end=4, label="person")
+ENTITY_TEXT_03_BERLIN = LabeledSpan(start=26, end=32, label="city")
+TEXT_03_SENTENCE1 = LabeledSpan(start=0, end=33, label="sentence")
+
+TEXT_01_TOKENS = [
+    "[CLS]",
+    "Jane",
+    "lives",
+    "in",
+    "Berlin",
+    ".",
+    "this",
+    "is",
+    "no",
+    "sentence",
+    "about",
+    "Karl",
+    "[SEP]",
+]
+TEXT_02_TOKENS = [
+    "[CLS]",
+    "Seattle",
+    "is",
+    "a",
+    "rainy",
+    "city",
+    ".",
+    "Jenny",
+    "Du",
+    "##rka",
+    "##n",
+    "is",
+    "the",
+    "city",
+    "'",
+    "s",
+    "mayor",
+    ".",
+    "[SEP]",
+]
+TEXT_03_TOKENS = ["[CLS]", "Karl", "enjoys", "sunny", "days", "in", "Berlin", ".", "[SEP]"]
+
+
+@pytest.fixture
+def candidate_relation_adder():
+    return CandidateRelationAdder(
+        label="no_relation",
+        collect_statistics=True,
+    )
```

Comment:
Is any of the fixtures (also below) used in multiple tests? If not, move them into the respective tests.

Reply:
There's separate for each, so I'll move them inside the test.